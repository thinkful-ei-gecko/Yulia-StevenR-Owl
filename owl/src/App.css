.App {
    display: flex;
    margin: 40px;
    min-height: calc(100vh - 80px);
  }

.stage {
    flex-basis: 70%;
    padding: 1em;
    background-color: grey;
}

.list {
    flex-basis: 10%;
    padding: 1em;
  }