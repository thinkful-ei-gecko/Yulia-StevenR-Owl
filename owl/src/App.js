import React from 'react';
import ParticipantList from './partList';
import Stage from './stage';
import Chat from './Chat';
import './App.css'

export default function App(props) {
    return (
        <main className="App">
            <ParticipantList className="list" participants={props.participants}></ParticipantList>
            <Chat className="list" chatEvents={props.chatEvents}></Chat>
            <Stage className="stage" participants={props.participants}></Stage>
        </main>
    )
};