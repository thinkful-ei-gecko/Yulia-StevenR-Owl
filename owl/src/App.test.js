import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import participants from './Store';
import chatEvents from './Chatdata';
import renderer from 'react-test-renderer';


describe('App', () => {
  it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');
  
    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<App participants={participants} chatEvents={chatEvents}/>, div);
  
    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<App participants={participants} chatEvents={chatEvents}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});