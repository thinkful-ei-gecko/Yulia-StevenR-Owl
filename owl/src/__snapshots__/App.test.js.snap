// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the UI as expected 1`] = `
<main
  className="App"
>
  <ul
    className="list"
  >
    <li>
      <h1>
        Koren Templeton
      </h1>
      <img
        alt="avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <span>
        on stage
      </span>
    </li>
    <li>
      <h1>
        Caty Flucker
      </h1>
      <img
        alt="avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <span>
        in session
      </span>
    </li>
    <li>
      <h1>
        Axe Kubicka
      </h1>
      <img
        alt="avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <span>
        left
      </span>
    </li>
    <li>
      <h1>
        Frank Runciman
      </h1>
      <img
        alt="avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <span>
        in session
      </span>
    </li>
    <li>
      <h1>
        Ashla Attwool
      </h1>
      <img
        alt="avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <span>
        on stage
      </span>
    </li>
  </ul>
  <ul
    className="list"
  >
    <li>
      <span>
        1/30/2019
      </span>
      <p>
        Hello world
      </p>
      <p>
        message
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        thumbs-up
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        thumbs-down
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        raise-hand
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        clap
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        join
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        leave
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        join-stage
      </p>
    </li>
    <li>
      <span>
        1/30/2019
      </span>
      <p />
      <p>
        leave-stage
      </p>
    </li>
  </ul>
  <section
    className="stage"
  >
    <span
      className="square"
    >
      <h1>
        Koren Templeton
      </h1>
      <img
        alt="participantsAvatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </span>
    <span />
    <span />
    <span />
    <span
      className="square"
    >
      <h1>
        Ashla Attwool
      </h1>
      <img
        alt="participantsAvatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </span>
  </section>
</main>
`;
